### PR TITLE
BLADE-377 init command should honor the -P, --profile-name flag over any defaults

### PR DESCRIPTION
**Pull request by christopherbryanboyd**
Hi @gamerson , this addresses https://issues.liferay.com/browse/BLADE-377

### CODE REVIEW

File: cli/src/main/java/com/liferay/blade/cli/BladeCLI.java [hunk 1]
```diff
@@ -401,6 +408,52 @@ private static String _extractBasePath(String[] args) {
 		return defaultBasePath;
 	}
 
+	private static String _extractProfileName(String[] args) {
+		String defaultProfile = null;
+
+		if (_isInitCommand(args) && (args.length > 2)) {
```
Comment (gamerson) [inline cli/src/main/java/com/liferay/blade/cli/BladeCLI.java]: Lets move all of this logic into the `InitCommand`.  I want to keep BladeCLI agnostic as to what commands are executing.  BladeCLI shouldn't "know" about InitCommand.

File: cli/src/main/java/com/liferay/blade/cli/BladeCLI.java [hunk 2]
```diff
@@ -401,6 +408,52 @@ private static String _extractBasePath(String[] args) {
 		return defaultBasePath;
 	}
 
+	private static String _extractProfileName(String[] args) {
+		String defaultProfile = null;
```
Comment (gamerson) [inline cli/src/main/java/com/liferay/blade/cli/BladeCLI.java]: Here you set defaultProfile to null and later you return default profile.  But nowhere do you set defaultProfile to something other than null, so it isn't of much use.

File: cli/src/main/java/com/liferay/blade/cli/Extensions.java [hunk 3]
```diff
@@ -387,6 +391,9 @@ private void _extractBladeExtensions(Path extensionsDirectory) throws IOExceptio
 
 						commandsToRemove.add(baseCommand);
 					}
+					else if (!profileNames.isEmpty() && !profileNames.contains(profileName)) {
```
Comment (gamerson) [inline cli/src/main/java/com/liferay/blade/cli/Extensions.java]: How is this block in 394 different or also needed when we have line 400-407?  Seems like those could be combined somehow?

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by christopherbryanboyd** [comment]
ci:retest

**Comment by christopherbryanboyd** [comment]
ci:retest

**Comment by christopherbryanboyd** [comment]
ci:retest

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/gamerson/liferay-blade-cli/pull/691. [See changes here.](https://github.com/gamerson/liferay-blade-cli/compare/34d83631~1...34d83631) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**